### PR TITLE
Support OPT_X_TLS_PEERCERT

### PR DESCRIPTION
**Pull request by tiran**
Replaces PR https://github.com/python-ldap/python-ldap/pull/401

### CODE REVIEW

**Comment by tiran** [comment]
Due to a bug in OpenLDAP, LDAP_OPT_X_TLS_PEERCERT leaks memory: https://bugs.openldap.org/show_bug.cgi?id=9696

File: Tests/t_ldapobject.py [hunk 1]
```diff
@@ -418,6 +423,36 @@ def test_multiple_starttls(self):
...
+        peercert = l.get_option(ldap.OPT_X_TLS_PEERCERT)
+        self.assertTrue(peercert)
+        self.assertIsInstance(peercert, bytes)
+
+        if PEM_cert_to_DER_cert is not None:
+            with open(self.server.servercert) as f:
+                server_pem = f.read()
+            # remove text
+            begin = server_pem.find("-----BEGIN CERTIFICATE-----")
+            server_pem = server_pem[begin:-1]
```
Comment (encukou) [inline Tests/t_ldapobject.py:451]: nitpick: Is removing the final character important? In our case itdoesn't really matter, but if the file doesn't end with a newline, the exception you get here is a bit obscure.
```suggestion
            server_pem = server_pem[begin:]
```

File: Tests/t_ldapobject.py [hunk 2]
```diff
@@ -421,6 +428,33 @@ def test_multiple_starttls(self):
             l.simple_bind_s(self.server.root_dn, self.server.root_pw)
             self.assertEqual(l.whoami_s(), 'dn:' + self.server.root_dn)
 
+    @requires_tls()
+    @unittest.skipUnless(
+        hasattr(ldap, "OPT_X_TLS_PEERCERT"),
```
Comment (mistotebe) [inline Tests/t_ldapobject.py:433]: Sounds like you still need to register this in `Lib/ldap/constants.py` to get the tests enabled?

**Comment by mistotebe** [comment]
Already done as part of #458

**Pull request closed without merging**